### ISSUE TITLE
C++ tools are not affected by sonar.exclusions property

### ISSUE THREAD

**Issue by zabulus** [SonarOpenCommunity/sonar-cxx-msbuild-tasks#65]
I'm trying to exclude our folder with 3rd party headers from analysis by C++ tools. 
We have a following structure in our project:
```/
/bin
/src
/src/project1
/src/projectN
/src/packages <-- folder I'd like to exclude from analysis
/src/packages/packagename/build/native/plenty/of/headers.hpp
```

I've tried so far:
1. sonar.exclusions in project settings
2. /d:sonar.exclusions in cli of cxxsonarquberunner
3. I've tried: ```**/packages/**/*```, ```packages/**/*```, ```packages/**/*.h```
4. I've tried to run runner from src, from root.
5. As last resort I've added ```/d:sonar.exclusions="d:\path\to\one\of\header\files\i\want\to\exclude.h"```

And only after 5th step I'm pretty sure that property ```sonar.exclusions``` doesn't work or not propagated to this method: https://github.com/jmecsoftware/CxxSonarQubeRunner/blob/540a505c083293367892f4d45d8e60e98ee077cb/CxxSonarQubeRunner/AnalysisRunners.fs#L48.

**Comment by jmecosta** [comment]
If u put in sonar runner props file? Does it work

**Comment by zabulus** [comment]
@jmecosta you mean xml file, that can be specified through ```/S``` CLI key?

**Comment by zabulus** [comment]
![image](https://user-images.githubusercontent.com/2720011/66561600-742ded00-eb62-11e9-8dcf-9d0c4c573ae9.png)
As I said properties from CLI are not propagated to this place. 
As for sonar.properties file. As I understand this file is auto-generated for each run. How can I patch it?

**Comment by zabulus** [comment]
Cause of not propagation is in:
https://github.com/jmecsoftware/CxxSonarQubeRunner/blob/481eda86cf259cf40ad4243c942c022ff85f2e2b/CxxSonarQubeRunner/Options.fs#L51
additionalArguments argument is not used.

**Comment by zabulus** [comment]
I've managed following patch. This will exclude folder pattern I'm interested in during analyzers phase.
```
 CxxSonarQubeRunner/Options.fs | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/CxxSonarQubeRunner/Options.fs b/CxxSonarQubeRunner/Options.fs
index 55f7d18..7ecccab 100644
--- a/CxxSonarQubeRunner/Options.fs
+++ b/CxxSonarQubeRunner/Options.fs
@@ -48,7 +48,7 @@ let GetPropertyFromFile(content : string [], prop : string) =
     | Some(c) -> c.Trim().Split('=').[1]
     | _ -> ""
 
-let GetArgumentClass(additionalArguments : string, content : string [], home : string, useCli:bool) = 
+let GetArgumentClass(additionalArguments : seq<string>, content : string [], home : string, useCli:bool) = 
     
     let mutable arguments = Map.empty
     
@@ -104,6 +104,7 @@ let GetArgumentClass(additionalArguments : string, content : string [], home : s
                 arguments <- arguments.Add(data.[0], data.[1])
 
     content |> Seq.iter (fun c -> ProcessLine(c.Trim()))
+    additionalArguments |> Seq.iter (fun c -> ProcessLine(c.Trim()))
 
     // ensure stuff that we run is included
     if not(arguments.ContainsKey("sonar.cxx.rats.reportPath")) then
@@ -796,7 +797,7 @@ type OptionsData(args : string []) =
         if options.UserSonarScannerCli then
             this.ConfigFile <- this.DeprecatedPropertiesFile
 
-        options.SonarPropsToUse <- GetArgumentClass(this.PropsForBeginStage, this.DepracatedSonarPropsContent, this.HomePath, options.UserSonarScannerCli)
+        options.SonarPropsToUse <- GetArgumentClass(arguments.["d"], this.DepracatedSonarPropsContent, this.HomePath, options.UserSonarScannerCli)
         if options.UserSonarScannerCli then
             WriteUserSettingsFromSonarPropertiesFileCli(this.ConfigFile, this.PropsInSettingsFile, options.SonarPropsToUse)
         else
```

**Comment by jmecosta** [comment]
https://docs.sonarqube.org/latest/analysis/scan/sonarscanner

Is was refering if u place those in that file instead of passing via cli /d
but of course for /d if that doesnt work feel free to do a pull request i
can review and merge...

**Comment by zabulus** [comment]
https://github.com/jmecsoftware/CxxSonarQubeRunner/pull/68/commits

**Comment by jmecosta** [comment]
Fix by #68 , thanks @zabulus  for the fix
